docs/model_developers/resource_sets.rst

```diff
@@ -480,6 +479,96 @@ behavior because the allocator guarantees uniqueness for the host id:
     {"Host(id=694)" "Host(id=269)" "Host(id=712)" "Host(id=31)"} -> "AgentConfig";
 
 
+Inter-resource set dependencies
+-------------------------------
+
+Resources within a resource set can only depend on resources within the same resource set or on shared resources.
+The diagram below provides an example where the resource dependency graph satisfies this requirement.
+
+.. digraph:: resource_sets_example_valid_dependencies
+    :caption: Two resource sets satisfying the dependency constraint
+
+    subgraph cluster_shared {
+        AgentConfig;
+        color = "green";
+        fontcolor = "grey";
+        label = "Shared resources";
+        labelloc = "bottom";
+    }
+
+    subgraph cluster_service0 {
+        "Network(id=0)" [shape=rectangle];
+        "Network(id=0)" -> subgraph cluster_resources0 {
+            "Host(id=269)";
+            color = "grey";
+            fontcolor = "grey";
+            label = "Resource set 0";
+            labelloc = "bottom";
+        }
+        color = "green";
+        label = "service 0";
+        labelloc = "top";
+    }
+    subgraph cluster_service1 {
+        "Network(id=1)" [shape=rectangle];
+        "Network(id=1)" -> subgraph cluster_resources1 {
+            "Host(id=31)";
+            color = "grey";
+            fontcolor = "grey";
+            label = "Resource set 1";
+            labelloc = "bottom";
+        }
+        color = "green";
+        label = "service 1";
+        labelloc = "top";
+    }
+
+    {"Host(id=269)" "Host(id=31)"} -> "AgentConfig";
+
+In the diagram below, resource ``Host(id=269)`` that belongs to resource set 0 depends on resource ``Host(id=31)`` that belongs to resource set 1. This inter-resource set dependency is not allowed.
+
+.. digraph:: resource_sets_example_invalid_dependencies
+    :caption: Two resource sets violating the dependency constraint
+
+    subgraph cluster_shared {
+        AgentConfig;
+        color = "green";
+        fontcolor = "grey";
+        label = "Shared resources";
+        labelloc = "bottom";
+    }
+
+    subgraph cluster_service0 {
+        "Network(id=0)" [shape=rectangle];
+        "Network(id=0)" -> subgraph cluster_resources0 {
+            "Host(id=269)";
+            color = "grey";
+            fontcolor = "grey";
+            label = "Resource set 0";
+            labelloc = "bottom";
+        }
+        color = "green";
+        label = "service 0";
+        labelloc = "top";
+    }
+    subgraph cluster_service1 {
+        "Network(id=1)" [shape=rectangle];
+        "Network(id=1)" -> subgraph cluster_resources1 {
+            "Host(id=31)";
+            color = "grey";
+            fontcolor = "grey";
+            label = "Resource set 1";
+            labelloc = "bottom";
+        }
+        color = "green";
+        label = "service 1";
+        labelloc = "top";
+    }
+
+    {"Host(id=269)" "Host(id=31)"} -> "AgentConfig";
+    {"Host(id=269)"} -> "Host(id=31)" [color=red];
```

Comment:
One minor visual thing: I feel the second diagram would look nicer if both service 0 and service 1 were aligned, but I don't know how difficult this would be. (Or if it would even actually look nicer in the end)
For reference: service1 is slightly below service0:
![image](https://user-images.githubusercontent.com/98876549/229468731-83e36f2c-e8ad-4319-b9ec-89c35feb8510.png)

Reply:
I tried to align them, but it seems that the feature to align them has a bug that causes all boxes in the clustergraph to disappear. So I am afraid we will have to keep it like it is, but I do agree it would be better to align them.